crates/bevy_state/src/app.rs

```diff
@@ -120,6 +132,14 @@ impl AppExtStates for SubApp {
 
         self
     }
+
+    #[cfg(feature = "bevy_hierarchy")]
+    fn enable_state_scoped_entities<S: States>(&mut self) -> &mut Self {
```

Comment:
This should warn if the app doesn't have `S` as a state already, or add it anyway

Reply:
Only warns, we have different methods for installing different state types